xplot.py

```diff
@@ -1 +1,49 @@
-#
+'''
+Problem 1: Plot Bar
+'''
+# TEST CASE GENERATOR # DO NOT MODIFY # 
+inputs_list = [["A", 627.2],["B", 38.9],["C", 11.3]]
+inputs_dict = {"A": 627.2, "B": 38.9,"C": 11.3}
+
+# SOLUTION
+
+def barGraph1(inputs_list):
+  max_val = max(inputs_list, key=lambda x: x[1])[1]
+  for element in inputs_list:
+      print(element[0],int((element[1]/max_val)*50)*"#")
+
+# Now, with a dict ...
+def barGraph2(inputs_dict):
+  max_val = max(inputs_dict.items(), key=lambda x: x[1])[1]
+  for key,val in inputs_dict.items():
+      print(key,int((val/max_val)*50)*"#")
+
+barGraph1(inputs_list)      
+barGraph1(inputs_dict)    
+
+    
+'''
+Problem 2: PlotX
+'''
+
+# TEST CASE GENERATOR # DO NOT MODIFY # 
+import random
+qty_points = random.randint(0,10)
+scale = random.randint(1,30)
+charmap = {1:"X ",0: ". "}
+input_coords = [[random.uniform(0.0,100.0) for _ in range(2)] for _ in range(qty_points)]   
+
+# SOLUTION
+def printGrid(input_coords,scale,charmap):
+    grid = [[charmap[0] for _ in range(scale)] for _ in range(scale)]
```

Comment:
`[[charmap[0]] * len(scale) for _ in range(scale)]` or you can also do, which I usually prefer, `defaultdictdict(lambda: '.')`

Reply:
This `[[charmap[0]] * len(scale) for _ in range(scale)]` won't work - will return type error since scale is an int and you can't take len(int). I think you meant perhaps: `[[charmap[0]] * scale for _ in range(scale)]` ? 

I like the approach with defaultdict(lambda: '.')
I understand this means when we call a key that does not exist, it will default to '. ' rather than return a key error.
However, we need to populate the grid, perhaps like grid[(a,b)] = "X" ? How do we use this dict to achieve the final printout?

Reply:
Yes you're right about scale.

For second question, you would do the same sort of thing as on line 45 but with a dict: grid[a][b] = 'X' (no need for parentheses). I guess a dict doesn't work well here (so matrix is better), unless you did something like defaultdict(lambda: ['.'] * scale), but that might be complicated.